.github/workflows/source-vendor.yml

```diff
@@ -0,0 +1,33 @@
+name: Create source archive with vendored dependencies
+
+on: [push, pull_request]
+
+jobs:
+  vendor-sources:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository with submodules
+        uses: actions/checkout@v4
+        with:
+          submodules: 'recursive'
+
+      - name: Create clean tarball
+        run: |
+          git archive --format=tar HEAD -o yosys-src-vendored.tar
+          git submodule foreach '
+            git archive --format=tar --prefix="${sm_path}/" HEAD --output=${toplevel}/vendor-${name}.tar
+          '
+          tar --concatenate --file=yosys-src-vendored.tar vendor-*.tar
+          # 2008 bug https://lists.gnu.org/archive/html/bug-tar/2008-08/msg00002.html
+          for file in vendor-*.tar; do
+              tar --concatenate --file=yosys-src-vendored.tar "$file"
+          done
+
+          gzip yosys-src-vendored.tar
+
+      - name: Store tarball artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: vendored-sources
+          path: yosys-src-vendored.tar.gz
+          retention-days: 1
```

Comment:
Is the idea here to replace the current ABC tarball with this combined tarball for inclusion in release?

Reply:
I don't know, but the availability of an all-inclusive tarball is kind of orthogonal to whether there is still a need to provide yosys-abc seperately in the yosys repo

Reply:
I guess the replacement is less what I was asking, and more is retention-days: 1 because the intention is for it to be added to releases, as opposed to being generally available for the most recent commit

Reply:
Yes, 1 because the intention is to be added to releases. My intention is to be able to run it manually actually, so:

- [ ]  re-add workflow_dispatch